### PR TITLE
Repo restructure project - Merge helm-locker into local

### PR DESCRIPTION
**Pull request by mallardduck**
This PR "merges in" the https://github.com/rancher/helm-locker project into this repo.

The code is simply dropped into `pkg/helm-locker` and all the other references are updated to reference the "local" module instead of the dedicated repo for it. Then the CI and related workflow scripts are all updated to account for having two CLI commands tested and compiled.

---

After this change the [open PR for updating helm-locker deps](https://github.com/rancher/helm-locker/pull/24/files#diff-0c3d932781349f0b60b198c9a1bc979fbbfad7b08865d28769892503c18cc2ea) can be reworked into this repo. At which time we will no longer need to update it in three locations before it lands. Granted it will still be needing updates here, and then subsequently in `prometheus-federated` repo - but two is still better than 3.

---

### Before we merge

- [x] Create a method for QA to test this before merge:
  - [x] Rework `prometheus-federator` to use this branch,
  - [x] Create version of charts that use these changes; provide chart branch and directions for install (verify w/ qa)
- [x] Verify the proper process for QA to test these changes (i.e. does installing my `prometheus-federator` chart and using it to deploy `rancher-project-monitoring` ensure they will test `helm-locker` fully in my fork?) [@jbiers will try to make progress on this while I'm out]
  - [x] Follow up with Pankaj on this process - I've already pinged him about the [issue here](https://github.com/rancher/prometheus-federator/issues/94). So may want to update that Issue with testing process info.
- [x] Confirm that `helm-locker` image isn't used standalone
  - [x] If yes, maybe update `hl-publish` workflow to only publish images to `ghcr.io` as they are for dev/QA only?
  - [x] Also adjust `hpo-publish` workflow similarly to push to `ghcr.io` - we know this is not user facing

> O&B team feel free to work on and merge this while I'm out if you have time and are comfortable picking it up.

### After we merge

- [ ] Tag a `v0.3.0` release here - which will release both images to GHCR
- [ ] Add `package/helm-project-operator` label to existing issues in this repo
- [ ] Move over the issues from `helm-locker` repo, label as `package/helm-locker`, archive `helm-locker` repo,
- [ ] Implement new branching strategy (maybe after Alex PR merge, need to decide optimal order)
- [ ] Rebuild Alex's PR to update helm-locker on this repo; will update both sets of dependencies at once.
- [ ] Update `prometheus-federator` to use patched versions of this repo,
- [ ] subsequently ensure `rancher-monitoring` and `rancher-project-monitoring` are updated and in charts

### CODE REVIEW

File: .gitignore
Comment (joshmeranda) [inline .gitignore:4]: why do we need to keep this dir? Can we recreated it when we need it?

File: charts/helm-locker/Chart.yaml
Comment (joshmeranda) [inline charts/helm-locker/Chart.yaml]: side note, at some point we might want to bring this more in line with the usual supported rancher version range, but this is super good enough for now

**Review by joshmeranda — APPROVED** [review]

**Review by jbiers — APPROVED** [review]
LGTM, read though all the commits and could not see any issues. However, it would be nice to wait for one more review before merging since another set of eyes could be useful in large PRs like this one.

File: .gitignore
Comment (mallardduck) [reply .gitignore:4]: Yeah - that's a fair point.
I tend to favor keeping dirs like this around - feels like it keeps things uniform and gives hints to the fact that the dir will be populated as part of a successful build. Not super attached to that though if we think it should just be fully excluded.

File: pkg/helm-locker/apis/helm.cattle.io/v1alpha1/doc.go
Comment (mallardduck) [reply pkg/helm-locker/apis/helm.cattle.io/v1alpha1/doc.go:1]: To make sure I understand the context - is this to address the concern about these as go pkgs? I.e. ensuring we properly migrate from: https://pkg.go.dev/github.com/rancher/helm-locker over to an equivalent?

**Pull request merged**
